oss-internship-2020/curl/tests/callbacks.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "callbacks.h"
+
+#include <cstdlib>
+#include <cstring>
+
+size_t write_to_memory(char* contents, size_t size, size_t nmemb, void* userp) {
+  size_t realsize = size * nmemb;
+  MemoryStruct* mem = (MemoryStruct*)userp;
+
+  char* ptr = (char*)realloc(mem->memory, mem->size + realsize + 1);
+  if (ptr == NULL) return 0;
```

Comment:
Use `nullptr` instead of `NULL`.

See https://google.github.io/styleguide/cppguide.html#0_and_nullptr/NULL